internal/ui/playlists.go

```diff
@@ -70,21 +62,21 @@ func (playlistModel PlaylistModel) View() string {
 	return docStyle.Render(playlistModel.list.View())
 }
 
-func NewPlaylistModel() PlaylistModel {
-	playlists := api.GetPlaylists()
+func NewPlaylistModel() *PlaylistModel {
+	var playlists []list.Item
 
-	for _, playlist := range playlists {
-		playlistItems = append(playlistItems, playlistItem{
-			id:   string(playlist.ID),
-			name: playlist.Name,
+	for _, p := range api.GetPlaylists() {
+		playlists = append(playlists, playlist{
+			ID:   string(p.ID),
+			Name: p.Name,
 		})
 	}
 
-	l := list.New(playlistItems, list.NewDefaultDelegate(), 0, 0)
+	l := list.New(playlists, list.NewDefaultDelegate(), 0, 0)
```

Comment:
Looking at my comment above, i'm not sure how would `list.New` deal with this in go, most probably it will just parse the array already by reference/pointer

Reply:
`func list.New(items []list.Item, delegate list.ItemDelegate, width int, height int) list.Model` I believe it needs the actual value here